### PR TITLE
[FIX] product: prevent getting value of domain from other method

### PR DESCRIPTION
**Pull request by prec-odoo**
While creating a new product in product category with barcode value in inventory module, `_check_barcode_uniqueness` method is called. In which variable 'domain' is passed, and getting wrong value.

Steps to Produce:-
1) Go to Inventory then configuration
2) Click on 'Product Categories' under products
3) Click/Create Product Category
4) Click stat button 'product'
5) Create a product
6) Then add barcode value
7) Click on Save
8) Trace-back will be generated

Reason:-

While creating new product from product category when user add the barcode value in product form, the '_check_barcode_uniqueness' method is called. In this method the value of 'domain' is getting updated by reference from variable 'domain' from method '_search' present in the same model. This results in a traceback with the message

'Invalid field product.packaging.categ_id'.

Applying these changes will resolve this issue.

See:-
ValueError-Invalid-field-product-packaging-categ_id-in-leaf-categ_id-child_of-6-online-saas

sentry - 4073963761

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by prec-odoo** [comment]
Hello @odoo/sentry_reviewers  Can you please look into this PR?

Thank you!

**Review by yelizariev — CHANGES_REQUESTED** [review]
I propose to make copy in `_search` method instead

**Comment by prec-odoo** [comment]
> I propose to make copy in `_search` method instead
> 
> https://github.com/odoo/odoo/blob/3e21dad7be482c7dbb9cd50619adf89c4b35cb8a/addons/product/models/product_product.py#L396-L401

Hello @yelizariev , I tried this but it didn't work. Thanks!

**Comment by yelizariev** [comment]
@prec-odoo can show what you have tried exactly?

**Comment by prec-odoo** [comment]
> @prec-odoo can you show what you have tried exactly?

@yelizariev , the solution has been updated. Can you please check the latest solution?

File: addons/product/models/product_product.py [hunk 1]
```diff
@@ -397,7 +397,7 @@ def copy(self, default=None):
     def _search(self, args, offset=0, limit=None, order=None, count=False, access_rights_uid=None):
         # TDE FIXME: strange
         if self._context.get('search_default_categ_id'):
-            args.append((('categ_id', 'child_of', self._context['search_default_categ_id'])))
+            args.copy().append((('categ_id', 'child_of', self._context['search_default_categ_id'])))
```
Comment (yelizariev) [inline addons/product/models/product_product.py:400]:
```suggestion
            args = args.copy()
            args.append((('categ_id', 'child_of', self._context['search_default_categ_id'])))
```

Comment (prec-odoo) [reply addons/product/models/product_product.py]: > 

@yelizariev , changes has been done.

**Comment by yelizariev** [comment]
@nle-odoo ?

**Review by nle-odoo — APPROVED** [review]
looks good to me

this is a small equivalent change, but that is fixing an error:

robodoo r+

File: addons/product/models/product_product.py [hunk 2]
```diff
@@ -397,6 +397,7 @@ def copy(self, default=None):
     def _search(self, args, offset=0, limit=None, order=None, count=False, access_rights_uid=None):
         # TDE FIXME: strange
         if self._context.get('search_default_categ_id'):
+            args = args.copy()
             args.append((('categ_id', 'child_of', self._context['search_default_categ_id'])))
```
Comment (adwid) [inline addons/product/models/product_product.py:401]:
```suggestion
            args = args + [('categ_id', 'child_of', self._context['search_default_categ_id'])]
```
Better for the perf (I know, it's a detail, but still)
robodoo delegate=yelizariev

**Pull request closed without merging**